### DISCUSSION TITLE
Green on green

### DISCUSSION THREAD

**Discussion by bassyboi** [geezacoleman/OpenWeedLocator#144] [Ideas]
Hi I put a an issue in the other day about the green on green!!! With coral and with pi since then I have installed coral using a python environment installing the version 3.9 on the environment!!! Now the trick is to install coral pi where it fails with the install is to use   This code python3 -m pip install --extra-index-url https://google-coral.github.io/py-repo/ pycoral~=2.0 it says it’s for Mac and pc on the document part of coral install page it’s the last part of the install… 

now I have the coral installed on my raspberrypi with 3.11 and the python environment with coral installed has 3.9!!! I have mucked around a little with the ini green on green!! Now I’m not super great at coding I can do some… how would I go about splitting the code up in the green on green parts that need to run in one environment and ones to run on (owl)

**Comment by bassyboi** [comment]
Also I do have a raspberry pi 5 with the ai top hat coming!!  Which was hard to find

**Reply by geezacoleman** [reply]
Yes, I have my own coming as well - still waiting though! I think this is the way forward and I would probably advise against persisting with the Coral unless you're very dedicated to getting it working. It's really unfortunate we haven't seen any moves nor support from Google in this space, it seems like such a big market. Would be great to get your experience with the AI Hat once it arrives - it's quite new to everyone, so we'll all probably come across similar issues.

**Reply by bassyboi** [reply]
i hope you don't mind i copied your repo in my own account im just going to add some changes so people can try... also i have some drawings for a boom spray for a lawn mower 515e john deere... thinking of doing a tractor boom soon too

**Reply by geezacoleman** [reply]
For sure - that's a good thing about GitHub is it makes forking a repo very easy so you can edit it to meet your own requirements. If there's anything you want to add to this main repository, you can then start a pull request and I can add it here.

Would be great to see how your boom sprays go - feel free to share the progress over at the Community Builds page #75. What solenoids are you using?

**Reply by bassyboi** [reply]
Plan at this stage is to get something small scale!!  A utv or something then maybe commercially available like a “weed it”retrofit it if the gog is good enough!!! 

I was talking to a farmer other than me that’s a neighbour of mine about price of solenoids so yes $230 dollars is expensive but isn’t the 5 grand+ a metre on the market!!! 

ATM im going to trial with MEIHE Water Solenoid Valve, 1pc DC 12V G3/4 Normal Closed Brass

**Reply by geezacoleman** [reply]
Fair point! The TeeJet e-chemsavers ones are quite decent and have very fast switching speeds. The [Goyen ](https://www.sprayerwarehouse.com.au/products/goyen-solenoid-1-2-bard-x-1-2-barb-straight?_pos=2&_sid=3d938601a&_ss=r)are another option too at around half the price.

**Reply by bassyboi** [reply]
yes i will check it out

**Comment by bassyboi** [comment]
https://github.com/bassyboi/yolo-models-for-weed-detection/tree/main/pasture%20and%20lawn

will make more models and train others working on a github page maybe if you started one id upload more as there trained... also trying to get some hailo models i can make onnx easily
